migrator/migrations/m_197_to_m_198_add_oidc_claim_mappings/migration_impl.go

```diff
@@ -0,0 +1,146 @@
+package m197tom198
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/migrator/migrations/m_197_to_m_198_add_oidc_claim_mappings/store/authproviders"
+	"github.com/stackrox/rox/migrator/migrations/m_197_to_m_198_add_oidc_claim_mappings/store/groups"
+	"github.com/stackrox/rox/migrator/types"
+	"github.com/stackrox/rox/pkg/auth/authproviders/oidc"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sac"
+)
+
+var (
+	log       = logging.LoggerForModule()
+	batchSize = 500
+)
+
+const (
+	orgIDKey            = "orgid"
+	accountIDClaim      = "account_id"
+	groupsKey           = "groups"
+	isOrgAdminKey       = "is_org_admin"
+	orgAdminGroupsValue = "org_admin"
+)
+
+func migrate(database *types.Databases) error {
+	ctx := sac.WithAllAccess(context.Background())
+	groupStore := groups.New(database.PostgresDB)
+	authProviderStore := authproviders.New(database.PostgresDB)
+	oidcAuthProviders := make([]*storage.AuthProvider, 0, batchSize)
+
+	err := authProviderStore.Walk(ctx, func(obj *storage.AuthProvider) error {
+		if obj.GetType() == oidc.TypeName {
+			oidcAuthProviders = append(oidcAuthProviders, obj)
+			if len(oidcAuthProviders) == batchSize {
+				if err := migrateAuthProviderClaims(ctx, groupStore, authProviderStore, oidcAuthProviders); err != nil {
+					return err
+				}
+				oidcAuthProviders = oidcAuthProviders[:0]
+			}
+		}
+		return nil
+	})
```

Comment:
Hmm, the auth provider store we have in `central/authprovider` uses [the get-all-func flag during generation](https://github.com/stackrox/stackrox/blob/master/central/authprovider/datastore/internal/store/postgres/gen.go#L3), let's include this in the store we have for the migration as well? Makes life a bit easier writing the migration + less error prone dealing with the batch sizes etc. since it's likely going to be a low-cardinality store anyways.

Reply:
The `GetAll` function is essentially a wrapper around `Walk` that puts all table entries in the list. We can add it or just remove the batching logic above.

Regarding batching: I got this idea by looking at https://github.com/stackrox/stackrox/blob/f2d669d1acc5f84ffe58c422f061850535638809/migrator/migrations/m_181_to_m_182_group_role_permission_with_access_one/migration_impl.go#L22-L66 -- another migration where you wouldn't expect too many entries. However, I think there's nothing wrong with being a little careful

Reply:
Personally think we can skip the batching for this one and use either the walk or introduce the GetAll to keep parity with the store in Central.